src/pkg/services/m365/api/lists.go

```diff
@@ -523,74 +500,35 @@ func CloneListItem(orig models.ListItemable) models.ListItemable {
 // additionalData map
 // Further documentation on FieldValueSets:
 // - https://learn.microsoft.com/en-us/graph/api/resources/fieldvalueset?view=graph-rest-1.0
-func retrieveFieldData(orig models.FieldValueSetable) models.FieldValueSetable {
+func retrieveFieldData(orig models.FieldValueSetable, columnNames map[string]struct{}) models.FieldValueSetable {
 	fields := models.NewFieldValueSet()
-	additionalData := filterAdditionalData(orig)
-
-	retainPrimaryAddressField(additionalData)
 
+	additionalData := setAdditionalDataByColumnNames(orig, columnNames)
 	fields.SetAdditionalData(additionalData)
 
 	return fields
 }
 
-func filterAdditionalData(orig models.FieldValueSetable) map[string]any {
+func setAdditionalDataByColumnNames(
+	orig models.FieldValueSetable,
+	columnNames map[string]struct{},
```

Comment:
prefer `any` over `struct{}`, please.  (we really should have a linter for this).
```suggestion
	columnNames map[string]any,
```